packages/frontend/src/features/semanticViewer/store/semanticViewerSlice.ts

```diff
@@ -329,11 +338,15 @@ export const semanticViewerSlice = createSlice({
             } else {
                 state.sortBy = [];
             }
-        },
 
+            state.shouldFetchResults = true;
+        },
         updateSaveModalOpen: (state, action: PayloadAction<boolean>) => {
             state.saveModalOpen = action.payload;
         },
+        setShouldFetchResults: (state, action: PayloadAction<boolean>) => {
+            state.shouldFetchResults = action.payload;
```

Comment:
hmm. im not super excited with this change. we had this in a previous explore and I can remember at least 3 times when we had to fix results not re-fetching after certain actions. I open to discuss how we can implement a "smart" auto re-fetch solution.

one of them being:
- any change that happens inside semantic query should run the refetch results:
- any change that happens inside semantic query and pivot config should refetch pivot results.

Reply:
Achieved this behavior by converting the hook into a `useQuery` with `sortBy` and `filters` on the query key. Also the hook is only enabled if any fields are selected and the user has clicked the run button 💪